digest.go

```diff
@@ -240,6 +242,15 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 		req2.Header[k] = s
 	}
 
+	// Copy body - we need it twice.
+	if req.Body != nil {
+		body, err := ioutil.ReadAll(req.Body)
+		if err != nil {
+			return nil, err
+		}
+		req.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+		req2.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+	}
```

Comment:
Reading the body into RAM isn't great if the payload is large. If req2.GetBody is non-nil we should use it to set req2.Body.

Reply:
`req2` is a shallow copy of `req`, so they both refer to the same reader.

Reply:
Yes, and that's the issue being solved in this PR, with the drawback that the body is buffered in RAM. By calling http.Request.GetBody on the request we can get a fresh io.Reader for the second request and not have to copy the data.

Reply:
Ah, that is some new fancy tech from Go 1.8 :D

You could also check if the reader is seekable... but this PR is dead anyway. Just leaving it here for others, not using it any more.

Reply:
Wow what a coincidence that this was discussed/revived in the last 24 hours. I spent 30 minutes debugging this problem and then an hour or two of research to find that this library had the bug of not cloning the Body. Most of the other clients I have seen do read the body into memory to clone it and none of them have been using `GetBody`. Any movement on this merge request? I've since switched to another library that is copying:
https://github.com/gabstv/httpdigest/blob/master/transport.go#L105

Reply:
Oh dear, _another_ digest package. I thought I'd found all of them yesterday. I'd say the chances of this PR getting merged are close to zero, but perhaps one of the other similar packages takes PRs. I've been using [github.com/toaster/digest](https://github.com/toaster/digest) myself. I'll probably send a PR to it for adding support for  http.Request.GetBody.

Reply:
@bobziuchkovski